libs/template/schema.go

```diff
@@ -0,0 +1,145 @@
+package template
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+)
+
+// This is a JSON Schema compliant struct that we use to do validation checks on
+// the provided configuration
+type Schema struct {
+	// A list of properties that can be used in the config
+	Properties map[string]Property `json:"properties"`
+}
+
+type PropertyType string
+
+const (
+	PropertyTypeString  = PropertyType("string")
+	PropertyTypeInt     = PropertyType("integer")
+	PropertyTypeNumber  = PropertyType("number")
+	PropertyTypeBoolean = PropertyType("boolean")
+)
+
+type Property struct {
+	Type        PropertyType `json:"type"`
+	Description string       `json:"description"`
+	Default     any          `json:"default"`
+}
+
+// function to check whether a float value represents an integer
+func isIntegerValue(v float64) bool {
+	return v == float64(int(v))
+}
+
+// cast value to integer for config values that are floats but are supposed to be
+// integers according to the schema
+//
+// Needed because the default json unmarshaler for maps converts all numbers to floats
+func castFloatConfigValuesToInt(config map[string]any, schema *Schema) error {
```

Comment:
This can be a member function on the schema.

Reply:
Since schema is now a common struct in `libs/schema`, I am leaving this as is. Let me know if this feedback still applies